core-web/src/main/java/org/silverpeas/core/webapi/base/HTTPAuthentication.java

```diff
@@ -0,0 +1,289 @@
+import org.silverpeas.core.admin.service.Administration;
+import org.silverpeas.core.admin.user.UserReference;
+import org.silverpeas.core.admin.user.model.UserDetail;
+import org.silverpeas.core.security.authentication.AuthenticationCredential;
+import org.silverpeas.core.security.authentication.AuthenticationService;
+import org.silverpeas.core.security.authentication.AuthenticationServiceProvider;
+import org.silverpeas.core.security.authentication.exception.AuthenticationException;
+import org.silverpeas.core.security.authentication.verifier.AuthenticationUserVerifierFactory;
+import org.silverpeas.core.security.session.SessionInfo;
+import org.silverpeas.core.security.session.SessionManagementProvider;
+import org.silverpeas.core.security.token.Token;
+import org.silverpeas.core.security.token.persistent.PersistentResourceToken;
+import org.silverpeas.core.util.Charsets;
+import org.silverpeas.core.util.Mutable;
+import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.util.logging.SilverLogger;
+import org.silverpeas.core.web.token.SynchronizerTokenService;
+
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.silverpeas.core.webapi.base.UserPrivilegeValidation.HTTP_ACCESS_TOKEN;
+import static org.silverpeas.core.webapi.base.UserPrivilegeValidation.HTTP_AUTHORIZATION;
+import static org.silverpeas.core.webapi.base.UserPrivilegeValidation.HTTP_SESSIONKEY;
+
+/**
+ * An HTTP authentication mechanism for Silverpeas. It implements the authentication mechanism
+ * in Silverpeas from an incoming HTTP request. This HTTP request can be as well an explicit
+ * authentication call as a Silverpeas API consume. The HTTP request is expected either to contain
+ * the HTTP header {@code Authorization} valued with the authentication scheme and the user
+ * credentials as expected by the IETF RFC 2617 or to target an URI with the query parameter
+ * {@code access_token} (see IETF RFC 6750).
+ * <p>
+ * Actually, Silverpeas supports two HTTP authentication schemes: the {@code Basic} one
+ * (covered by the IETF RFC 2617) and the Bearer one (covered by the IETF RFC 6750). The API token
+ * of the users must be passed with the {@code Bearer} scheme to access the REST API of
+ * Silverpeas.
+ * </p>
+ * <p>
+ * The authentication opens a new session when succeeded, otherwise a
+ * {@link WebApplicationException} exception is thrown with the status
+ * {@link Response.Status#UNAUTHORIZED}.
+ * </p>
+ * @author mmoquillon
+ */
+@Singleton
+public class HTTPAuthentication {
+
+  private static Pattern AUTHORIZATION_PATTERN = Pattern.compile("(?i)^(Basic|Bearer) (.*)");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "AUTHORIZATION_PATTERN" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3008)